internal/app/frontend/frontend_service_test.go

```diff
@@ -88,6 +88,62 @@ func TestDoCreateTickets(t *testing.T) {
 	}
 }
 
+func TestDoCreateBackfill(t *testing.T) {
```

Comment:
While I think this follows the patterns of the testing for the other calls, I'm unsure about the value these tests are adding:

- I don't think calling with canceled context actually finds problems, at most if checks that one use of context is done correctly, but not that all uses are correct.
- It took a bit to get there, but I'm pretty confident that "preAction" is an anti-pattern at this point.  It's adding too much logic to tests.
- See also, the if err != nil check in the test code is definitely logic.  This should be removed, and is evidence of the tests being over-generalized.

My recommendation:
Get the test coverage using the e2e tests for:
- creating
- getting backfill created
- getting backfill which isn't present, and getting not found
(first two are already done, as far as I can tell)

At most, keep only the canceled context tests here, but I'm fine not having those.
The other tests that follow these patterns for existing features should probably be cleaned up, but that's out of scope for this project.

Reply:
Thanks for your feedback, will address all comments